test/quick_test/test_images.py

```diff
@@ -0,0 +1,219 @@
+        reference_sum = np.sum(reference_data == current_label)
+        test_sum = np.sum(test_data == current_label)
+        intersection = np.sum((reference_data == current_label) & (test_data == current_label))
+
+        if intersection == 0 and classes == 2:
+            if current_label == 17:
+                swap_lab = 53
+            else:
+                swap_lab = 17
+
+            reference_sum = np.sum(reference_data == swap_lab)
+            intersection = np.sum((reference_data == swap_lab) & (test_data == current_label))
+
+        dice_score[idx] = (2 * intersection) / (reference_sum + test_sum)
+
+    dice_score[np.isnan(dice_score)] = 0.
+
+    print("\nDice score: ", dice_score)
+
+    return dice_score
+
+
+def compute_mean_square_error(test_data, reference_data):
+    """
+    Calculate the mean square error.
+
+    Parameters
+    ----------
+    test_data : np.ndarray
+        Test image data.
+    reference_data : np.ndarray
+        Reference image data.
+
+    Returns
+    -------
+    float
+        Mean square error.
+    """
+
+    mse = ((test_data - reference_data) ** 2).mean()
+    print("\nMean square error: ", mse)
+
+    return mse
+
+
+def test_image_headers(sample_subjects_dir, sample_subject_id, sample_reference_id):
+    """
+    Test image headers using nibabel.
+
+    Parameters
+    ----------
+    sample_subjects_dir : str
+        Path to the sample subjects directory.
+    sample_subject_id : str
+        Sample subject id.
+    sample_reference_id : str
+        Sample reference id.
+    """
+
+    # Load images
+    test_image = load_image(sample_subjects_dir, sample_subject_id, "brain.mgz")
+    reference_image = load_image(sample_subjects_dir, sample_reference_id, "brain.mgz")
+
+    # Get the image headers
+    headers = [test_image.header, reference_image.header]
+
+    # Check the image header
+    assert nibabel.cmdline.diff.get_headers_diff(headers) == OrderedDict(), "Image headers do not match"
+    print("Image headers are correct")
+
+
+def test_seg_data(sample_subjects_dir, sample_subject_id, sample_reference_id):
+    """
+    Test the segmentation data by calculating and comparing dice scores.
+
+    Parameters
+    ----------
+    sample_subjects_dir : str
+        Path to the sample subjects directory.
+    sample_subject_id : str
+        Sample subject id.
+    sample_reference_id : str
+        Sample reference id.
+    """
+
+    test_image = load_image(sample_subjects_dir, sample_subject_id, "brain.mgz")
+    reference_image = load_image(sample_subjects_dir, sample_reference_id, "brain.mgz")
+
+    # Get the image data
+    test_data = test_image.get_fdata()
+    reference_data = reference_image.get_fdata()
+
+    # Compute the dice score
+    dice_score = compute_dice_score(test_data, reference_data)
+
+    # Check the dice score
+    for score in dice_score:
+        assert score == 1, f"Dice score is not 1 for f{np.where(dice_score == score)[0][0]}"
```

Comment:
multiple asserts per call, see if this can be made into one assert (either by splitting into multiple functions or by combining the checks via `any()`.